doc/c-style-guide.md

```diff
@@ -0,0 +1,41 @@
+GRPC C STYLE GUIDE
+=====================
+
+Background
+----------
+
+Here we document style rules for C usage in the gRPC Core library.
+
+General
+-------
+
+- Layout rules are defined by clang-format, and all code should be passed through
+  clang-format. A (docker-based) script to do so is included in 
+  tools/distrib/clang_format_code.sh.
+
+Header Files
+------------
+
+- Public header files (those in the include/grpc tree) should compile as pedantic C89
```

Comment:
So I don't think anyone is at exactly c89, but I don't think the world is all c99 either. Particularly there are wrapping toolkits that don't necessarily provide great implementations of the language, and I'd like to keep it easy for folks using them. Even if parts of the census C api are intended to be used directly from C++, they'll need wrapping for our other six languages, and hopefully whichever languages the community provides wrappings for. Since I can't know all the restrictions that might be imposed, I feel for this part it's important to be as conservative as possible and choose a machine verifiable subset of the language. Pedantic C89 has served us wonderfully there.